source/extensions/filters/network/dubbo_proxy/app_exception.cc

```diff
@@ -0,0 +1,41 @@
+#include "extensions/filters/network/dubbo_proxy/app_exception.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/network/dubbo_proxy/message.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+AppException::AppException(ResponseStatus status, const std::string& what)
+    : EnvoyException(what), status_(status),
+      response_type_(RpcResponseType::ResponseWithException) {}
+
+AppException::AppException(const AppException& ex)
+    : EnvoyException(ex.what()), status_(ex.status_) {}
+
+AppException::ResponseType AppException::encode(MessageMetadata& metadata,
+                                                DubboProxy::Protocol& protocol,
+                                                Deserializer& deserializer,
+                                                Buffer::Instance& buffer) const {
+  ENVOY_LOG(debug, "err {}", what());
+
+  metadata.setResponseStatus(status_);
+  metadata.setMessageType(MessageType::Response);
+
+  const std::string& response = what();
+  if (!protocol.encode(buffer, response.size(), metadata)) {
```

Comment:
response.size() equal serializeRpcResult(buffer, response, response_type_) ?

Reply:
I've fixed it, body size = response.size() + sizeof(response_type_).

Reply:
`body_size` not equal `response.size() + sizeof(response_type_)`

Reply:
I've fixed it.